contracts/contracts/token/OUSD.sol

```diff
@@ -584,20 +633,92 @@ contract OUSD is Initializable, InitializableERC20Detailed, Governable {
             ? MAX_SUPPLY
             : _newTotalSupply;
 
-        _rebasingCreditsPerToken = _rebasingCredits.divPrecisely(
-            _totalSupply.sub(nonRebasingSupply)
-        );
+        _rebasingCreditsPerToken =
+            (_rebasingCredits * 1e18) /
+            (_totalSupply - nonRebasingSupply);
 
         require(_rebasingCreditsPerToken > 0, "Invalid change in supply");
 
-        _totalSupply = _rebasingCredits
-            .divPrecisely(_rebasingCreditsPerToken)
-            .add(nonRebasingSupply);
+        _totalSupply =
+            ((_rebasingCredits * 1e18) / _rebasingCreditsPerToken) +
+            nonRebasingSupply;
 
         emit TotalSupplyUpdatedHighres(
             _totalSupply,
             _rebasingCredits,
             _rebasingCreditsPerToken
         );
     }
+
+    function delegateYield(address from, address to)
+        external
+        onlyGovernor
+        nonReentrant
+    {
+        require(from != to, "Cannot delegate to self");
+        require(
+            yieldFrom[to] == address(0) &&
+                yieldTo[to] == address(0) &&
+                yieldFrom[from] == address(0) &&
+                yieldTo[from] == address(0),
+            "Blocked by existing yield delegation"
+        );
+        RebaseOptions stateFrom = rebaseState[from];
+        RebaseOptions stateTo = rebaseState[to];
+        if (!_isNonRebasingAccount(from) &&
+                (stateFrom == RebaseOptions.NotSet ||
+                    stateFrom == RebaseOptions.StdRebasing)) {
+            _rebaseOptOut(from);
+        }
+        if(_isNonRebasingAccount(to) &&
+                (stateTo == RebaseOptions.NotSet ||
+                    stateTo == RebaseOptions.StdNonRebasing)) {
+            _rebaseOptIn(to);
```

Comment:
@DanielVF any objection for the yield delegation function to force the rebase receiving account to also be forced into rebaseOptIn?
https://github.com/OriginProtocol/origin-dollar/pull/2298/commits/ef7c4ab26b56f38ff669de8c151eb4f4a0ae2285

Reply:
It's fine do. Will comment on the specific implementation in another comment.